solidity/security/balancer-readonly-reentrancy-getpooltokens.yaml

```diff
@@ -0,0 +1,54 @@
+rules:
+    - id: balancer-readonly-reentrancy-getpooltokens
+      message: Use VaultReentrancyLib to protect from readonly reentrancy.
+      metadata: 
+        references: 
+          - https://quillaudits.medium.com/decoding-sentiment-protocols-1-million-exploit-quillaudits-f36bee77d376
+        category: security
+        tags:
+          - reentrancy
+          - balancer
+      patterns:
+        - pattern-either:
+          - pattern: |
+              $RETURN = $VAULT.getPoolTokens(...)
+        - metavariable-pattern:
+              metavariable: $RETURN
+              pattern-regex: .*uint256\[].*
+        - pattern-not-inside: |
```

Comment:
Sentiment fixed `includingWeightedBalancerLPOracle` the following way in https://arbiscan.io/address/0xC0Fc3193Bf2176D1DA6D2F24C14996766f46eB67#code:

```
function getPrice(address token) external returns (uint) {
        checkReentrancy();
        ...
}
function checkReentrancy() internal {
      vault.manageUserBalance(new IVault.UserBalanceOp[](0));
  }
```
More info here: https://hackmd.io/@sentimentxyz/SJCySo1z2